From 5fcabe74650ca269eb189458c6e964146a6785f6 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Sun, 10 Apr 2022 20:45:51 +0300
Subject: [PATCH] [SW-77251] gaudi3: skip rom recovery for slave die

ROM recovery is expected to be performed only by master die, this
patch add change to skip rom recovery for slave die.

Change-Id: Ia7bbb37cf4dc70d7dce30bea48ddcdd5ce524c56
---

diff --git a/arch/gaudi3/arc/recovery.c b/arch/gaudi3/arc/recovery.c
index 3680228..161422c 100644
--- a/arch/gaudi3/arc/recovery.c
+++ b/arch/gaudi3/arc/recovery.c
@@ -155,6 +155,10 @@
 
 static void rom_recovery(void)
 {
+	/* ROM recovery not supported in slave die */
+	if (arcboot_rom_global.die_id)
+		return;
+
 	device_init();
 
 	module_handle_cmds(cmd_ops, ARRAY_SIZE(cmd_ops));
